app/scripts/app.config.js

```diff
@@ -219,6 +220,36 @@ angular
           },
         },
       })
+      .when('/paymentReport/:conferenceId', {
+        title: gettext('Payment Report Preview'),
+        templateUrl: paymentReportTemplate,
+        controller: 'paymentReportCtrl',
```

Comment:
Would it mean that we need the new parent component with its own routing?

Reply:
You can leave it since it would be a little bit of a rewrite. Since this app is very dependant on shared scopes (which is so messy) it might be hard to start isolating things but it would be a lot nicer.

But ya components are nice since they enforce some of the best practices from directives, keep everything grouped together, and have isolate scope. https://docs.angularjs.org/guide/component

Here are some [example components from another project](https://github.com/CruGlobal/missionhub-web/tree/master/app/assets/javascripts/angular/components) and the [router config](https://github.com/CruGlobal/missionhub-web/blob/master/app/assets/javascripts/angular/missionhubApp.routes.js#L245).